apps/full-site-editing/full-site-editing-plugin/premium-content/index.js

```diff
@@ -33,11 +39,40 @@ const registerBlock = ( block ) => {
 	} );
 };
 
+/**
+ * Appends a "paid" tag to the Premium Content block title if site requires an upgrade.
+ */
+const addPaidBlockFlags = async () => {
+	const membershipsStatus = await apiFetch( { path: '/wpcom/v2/memberships/status' } );
+	const shouldUpgrade = membershipsStatus.should_upgrade_to_access_memberships;
+	if ( shouldUpgrade ) {
+		const premiumContentBlock = getBlockType( container.name );
+		if ( ! premiumContentBlock ) {
+			return;
+		}
+
+		const paidFlag = _x(
+			'paid',
+			'Short label appearing near a block requiring a paid plan',
+			'premium-content'
+		);
+
+		unregisterBlockType( container.name );
```

Comment:
I wish there was some other way of updating the settings of a block already registered.